### DISCUSSION TITLE
Slight restructure?

### DISCUSSION THREAD

**Discussion by prologic** [NicoNex/tau#16] [Ideas]
Do you mind if we restructure this project a bit?

- Move the `*.go` in the top-level dir to `./cmd/tau/`
- Introduce a `tau.go` in the top-level dir with an exported `Main()` like the one in my [ed project](https://git.mills.io/prologic/ed/src/branch/master/main.go)?

This will allow me to easily embed `taue` into my project where I basically have an entirely Go-written userland busybox style (single-binary) 😅 An applet basically has teh form:

```#!go
package template

import (
	"flag"
	"fmt"
	"io"

	"git.mills.io/prologic/gonix/internal/applets"
)

const binaryName = "template"

const helpText = `
This is a template applet.
This binary does nothing.
`

func init() {
	applets.Register(binaryName, Main)
}

type Option struct {
	helpFlag bool
}

func NewFlagSet() (*flag.FlagSet, *Option) {
	ret := flag.NewFlagSet(binaryName, flag.ExitOnError)

	ret.Usage = func() {
		fmt.Printf("Usage: %s [OPTIONS] ...\n", binaryName)
		fmt.Printf("%s\n", helpText)
		ret.PrintDefaults()
	}

	var opt Option

	ret.BoolVar(&opt.helpFlag, "help", false, "show this message")

	return ret, &opt
}

func Main(stdout io.Writer, args []string) error {
	flagSet, opt := NewFlagSet()
	flagSet.Parse(args)

	//if flagSet.NArg() < 1 || opt.helpFlag {
	if opt.helpFlag {
		flagSet.Usage()
		return nil
	}

	return template(stdout, flagSet.Args(), opt)
}

func template(w io.Writer, args []string, opt *Option) error {
	return nil
}
```

**Comment by NicoNex** [comment]
Hi @prologic, what do you need to do exactly? And what's wrong with the current layout?

**Reply by prologic** [reply]
It's okay. I'll fork and do this and show you 👌

**Reply by NicoNex** [reply]
Ok, btw sorry for the slow responses but unfortunately I got covid and so I don't have much energy to work lately.

**Reply by prologic** [reply]
Sorry to hear that 😢
